adapters/Pangle/PangleAdapter/GADMediationAdapterPangle.m

```diff
@@ -29,13 +31,19 @@ @implementation GADMediationAdapterPangle {
   GADPangleRTBInterstitialRenderer *_interstitialRenderer;
   /// Pangle rewarded ad wrapper.
   GADPangleRTBRewardedRenderer *_rewardedRenderer;
+  /// Pangle native ad wrapper.
+  GADPangleRTBNativeRenderer *_nativeRenderer;
 }
 
 - (void)collectSignalsForRequestParameters:(nonnull GADRTBRequestParameters *)params
                          completionHandler:
-                             (nonnull GADRTBSignalCompletionHandler)completionHandler {
-  NSString *signals = [BUAdSDKManager getBiddingToken:nil];
-  completionHandler(signals, nil);
+(nonnull GADRTBSignalCompletionHandler)completionHandler {
+    GADPangleNetworkExtras *extras = [params.extras isKindOfClass:[GADPangleNetworkExtras class]] ? params.extras : nil;
+    if (extras && extras.userDataString.length > 0) {
```

Comment:
Just `if (extras.userDataString.length)` should be sufficient? Please double check.